src/shared-validator.cc

```diff
@@ -479,8 +510,11 @@ Result SharedValidator::OnLocalDecl(const Location& loc,
     PrintError(loc, "local count must be < 0x10000000");
     return Result::Error;
   }
+
+  Result result = CheckReferenceType(loc, type, "locals");
```

Comment:
Use `CHECK_RESULT` here and revert line 483?

Reply:
Good question. What happens if the variable is not added to `locals_`?

Reply:
I think if we return error then all best are off, no?  Should be fine I think

Reply:
Ok, I changed it to CHECK_RESULT.